src/rbr/RBR.java

```diff
@@ -131,22 +131,26 @@ public void regionsComputation() {
 
     private void adjustRegions(Vertex sw) {
         List<Region> newRegions = new ArrayList<>();
-        List<Region> regionsToBeRemoved = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            Set<Vertex> outsiders = currentRegion.outsiders();
-            if(outsiders.isEmpty())
-                continue;
+            newRegions.addAll(adjustedRegions(currentRegion));
+        }
+        regionsForVertex.put(sw, newRegions);
+    }
+
+    private ArrayList<Region> adjustedRegions(Region r) {
+        ArrayList<Region> newRegions = new ArrayList<>();
```

Comment:
I'm sorry. I didn't make myself clear enough. 'adjustedRegionsFrom' was a very good name to that method. What I asked to change was the variable 'newRegions' which I think could be named 'adjustedRegions'